extensions/azure/data-plane/data-factory/src/test/java/org/eclipse/dataspaceconnector/azure/dataplane/azuredatafactory/AzureDataFactoryCopyIntegrationTest.java

```diff
@@ -145,6 +162,23 @@ void transfer_success(
                 .isEqualTo(randomBytes.length);
     }
 
+    private void setSecret(Account consumerStorage, Vault vault, String secretName) {
+        // ADF SLA to start an activity is 4 minutes.
+        var expiryTime = OffsetDateTime.now().plusMinutes(8);
+        var permission = new BlobContainerSasPermission().setWritePermission(true);
+        var sasSignatureValues = new BlobServiceSasSignatureValues(expiryTime, permission)
+                .setStartTime(OffsetDateTime.now());
+        var sasToken = consumerStorage.client
+                .getBlobContainerClient(consumerStorage.containerName)
+                .generateSas(sasSignatureValues);
+        var edcAzureSas = new AzureSasToken(sasToken, expiryTime.toEpochSecond());
+        // Set Secret
+        vault.secretClient().setSecret(secretName, typeManager.writeValueAsString(edcAzureSas)).block(Duration.ofMinutes(1));
+        // Add for clean up test data
+        secretCleanup.add(() -> vault.secretClient().beginDeleteSecret(secretName).blockLast(Duration.ofMinutes(1)));
+        secretCleanup.add(() -> vault.secretClient().purgeDeletedSecret(secretName).block(Duration.ofMinutes(1)));
```

Comment:
why we need these 2 methods beginDeleteSecret and purgeDeletedSecret ?

Reply:
It is because we have soft delete enabled on key vault so if I just delete the key then we will not be able to recreate it with the same name I think for 7 days as it is not purged and eligible for recovery. So, need to purge it as well.